File: app/src/main/java/com/example/module5task1java/MainActivity.java
package com.example.module5task1java;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.viewpager.widget.ViewPager;
import androidx.viewpager2.widget.ViewPager2;

import com.tbuonomo.viewpagerdotsindicator.DotsIndicator;
import com.tbuonomo.viewpagerdotsindicator.WormDotsIndicator;

import me.relex.circleindicator.CircleIndicator3;

public class MainActivity extends AppCompatActivity {

    ViewPager viewPager;
    CustomPagerAdapter adapter;
    WormDotsIndicator indicator;
    TextView textView;
    Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();
    }

    private void initViews() {

        textView = findViewById(R.id.textView);
        button = findViewById(R.id.button);
        viewPager = findViewById(R.id.viewPager);
        indicator = findViewById(R.id.worm_dots_indicator);

        adapter = new CustomPagerAdapter(getSupportFragmentManager());
        viewPager.setAdapter(adapter);
        indicator.setViewPager(viewPager);

        textView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                viewPager.setCurrentItem(2, true);
                button.setVisibility(View.VISIBLE);
            }
        });

        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                if(position != 2){
                    button.setVisibility(View.INVISIBLE);
                } else {
                    button.setVisibility(View.VISIBLE);
                }
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });

    }
}

File: app/src/main/java/com/example/module5lesson1task1kotlin/MainActivity.kt
package com.example.module5lesson1task1kotlin

import android.graphics.Color
import android.os.Bundle
import android.text.SpannableString
import android.text.Spanned
import android.text.TextPaint
import android.text.method.LinkMovementMethod
import android.text.style.BackgroundColorSpan
import android.text.style.ClickableSpan
import android.text.style.ForegroundColorSpan
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.tbuonomo.viewpagerdotsindicator.WormDotsIndicator

class MainActivity : AppCompatActivity() {

    lateinit var viewPager: ViewPager
    lateinit var adapter: CustomPagerAdapter
    lateinit var indicator: WormDotsIndicator
    lateinit var textView: TextView
    lateinit var button: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()
    }

    private fun initViews() {
        textView = findViewById(R.id.textView)
        button = findViewById(R.id.button)
        viewPager = findViewById(R.id.viewPager)
        indicator = findViewById(R.id.worm_dots_indicator)
        adapter = CustomPagerAdapter(supportFragmentManager)

        viewPager.setAdapter(adapter)
        indicator.setViewPager(viewPager)

        val ssString = SpannableString(textView.getText())
        val bcs = BackgroundColorSpan(Color.WHITE)
        val fcs = ForegroundColorSpan(Color.RED)

        val clickableSpan: ClickableSpan = object : ClickableSpan() {
            override fun onClick(view: View) {
                viewPager.setCurrentItem(2, true)
                button.setVisibility(View.VISIBLE)
            }

            override fun updateDrawState(ds: TextPaint) {
                ds.color = Color.BLUE
                ds.isUnderlineText = true
            }
        }
        ssString.setSpan(bcs, 0, textView.getText().length, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
        ssString.setSpan(
            clickableSpan,
            0,
            textView.getText().length,
            Spanned.SPAN_EXCLUSIVE_EXCLUSIVE
        )
        textView.setText(ssString)
        textView.setMovementMethod(LinkMovementMethod.getInstance())
        viewPager.addOnPageChangeListener(object : OnPageChangeListener {
            override fun onPageScrolled(
                position: Int,
                positionOffset: Float,
                positionOffsetPixels: Int
            ) {
            }

            override fun onPageSelected(position: Int) {
                if (position != 2) {
                    button.setVisibility(View.INVISIBLE)
                } else {
                    button.setVisibility(View.VISIBLE)
                }
            }

            override fun onPageScrollStateChanged(state: Int) {}
        })
    }
}